homeassistant/helpers/script.py

```diff
@@ -269,41 +273,55 @@ async def _async_delay_step(self):
 
     async def _async_wait_template_step(self):
         """Handle a wait template."""
+        try:
+            delay = self._get_pos_time_period_template(CONF_TIMEOUT).total_seconds()
+        except KeyError:
```

Comment:
Let's check for `CONF_TIMEOUT` instead of catching key error, because we call a lot of code and we wouldn't want to accidentally catch unrelated key errors.